src/Analyzers/CSharp/Tests/ConstantInterpolatedString/ConstantInterpolatedStringTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.ConstantInterpolatedString;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.UnitTests.ConstantInterpolatedString
+{
+    using VerifyCS = CSharpCodeFixVerifier<CSharpConstantInterpolatedStringAnalyzer, EmptyCodeFixProvider>;
+
+    public sealed class ConstantInterpolatedStringTests
+    {
+        [Fact]
+        public async Task TestSimpleSingleConcatenation()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+public class C
+{
+    public void M()
+    {
+        const string x = [|""a"" + ""b""|];
```

Comment:
After a second look, I think this is unrelated and shouldn't produce a diagnostic. So we may only need a diagnostic only if one of the operands isn't a string literal.

@sharwell Can you confirm?